src/appleseed-max-impl/appleseedblendmtl/appleseedblendmtl.cpp

```diff
@@ -0,0 +1,744 @@
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2017 Sergo Pogosyan, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "appleseedblendmtl.h"
+
+// appleseed-max headers.
+#include "appleseedblendmtl/datachunks.h"
+#include "appleseedblendmtl/resource.h"
+#include "appleseedrenderer/appleseedrenderer.h"
+#include "bump/bumpparammapdlgproc.h"
+#include "bump/resource.h"
+#include "main.h"
+#include "oslutils.h"
+#include "utilities.h"
+#include "version.h"
+
+// appleseed.renderer headers.
+#include "renderer/api/material.h"
+#include "renderer/api/scene.h"
+#include "renderer/api/shadergroup.h"
+#include "renderer/api/utility.h"
+#include "renderer/modeling/shadergroup/shader.h"
+#include "renderer/modeling/shadergroup/shaderconnection.h"
+
+// appleseed.foundation headers.
+#include "foundation/image/colorspace.h"
+
+// 3ds Max headers.
+#include <color.h>
+#include <iparamm2.h>
+#include <stdmat.h>
+#include <strclass.h>
+
+// Standard headers.
+#include <string>
+
+// Windows headers.
+#include <tchar.h>
+
+namespace asf = foundation;
+namespace asr = renderer;
+
+namespace
+{
+    const wchar_t* AppleseedBlendMtlFriendlyClassName = L"appleseed Blend Material";
+}
+
+AppleseedBlendMtlClassDesc g_appleseed_blendmtl_classdesc;
+
+
+//
+// AppleseedBlendMtl class implementation.
+//
+
+namespace
+{
+    enum { ParamBlockIdBlendMtl };
+    enum { ParamBlockRefBlendMtl };
+
+    enum ParamId
+    {
+        // Changing these value WILL break compatibility.
+        ParamIdBaseMtl          = 0,
+        ParamIdLayerMtl         = 1,
+        ParamIdMaskTex          = 2,
+        ParamIdMaskAmount       = 3,
+        ParamIdMixColor         = 4,
+    };
+
+    enum MtlId
+    {
+        // Changing these value WILL break compatibility.
+        MaterialBase                = 0,
```

Comment:
Align `=` vertically.

Reply:
You started counting at 0 in the OSL shader but you're counting from 1 here. It would make sense to always count similarly.

Reply:
Ok, I'll rename osl inputs to 1..10

Reply:
You might find it more pleasing to go with 9 "coating" materials so that things stay aligned :)

Reply:
Agree, number 10 is very annoying :)